airbyte-integrations/connectors/source-s3/source_s3/v4/unstructured_parser.py

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+import logging
+from io import IOBase
+from typing import Any, Dict, Iterable, List, Mapping, Optional
+
+from airbyte_cdk.sources.file_based.config.file_based_stream_config import FileBasedStreamConfig
+from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader, FileReadMode
+from airbyte_cdk.sources.file_based.file_types.file_type_parser import FileTypeParser
+from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+from airbyte_cdk.sources.file_based.schema_helpers import SchemaType
+from source_s3.v4.config import S3FileBasedStreamConfig
+
+
+class UnstructuredParser(FileTypeParser):
+
+    MAX_SIZE_PER_CHUNK = 4_000_000
+
+    async def infer_schema(
+        self,
+        config: S3FileBasedStreamConfig,
+        file: RemoteFile,
+        stream_reader: AbstractFileBasedStreamReader,
+        logger: logging.Logger,
+    ) -> SchemaType:
+        return {
+            "content": {"type": "string"},
+            "chunk_number": {"type": "integer"},
+            "no_of_chunks": {"type": "integer"},
+            "id": {"type": "string"},
+        }
+
+    def parse_records(
+        self,
+        config: FileBasedStreamConfig,
+        file: RemoteFile,
+        stream_reader: AbstractFileBasedStreamReader,
+        logger: logging.Logger,
+        discovered_schema: Optional[Mapping[str, SchemaType]],
+    ) -> Iterable[Dict[str, Any]]:
+        with stream_reader.open_file(file, self.file_read_mode, None, logger) as fp:
+            markdown = self.read_file(fp, logger)
+            if not markdown:
+                return []
```

Comment:
To keep the return type of this consistent, this should return an iterator instead of an iterable, e.g. `yield from []`.

Reply:
Got removed when removing chunks